bukuserver/api.py

```diff
@@ -101,16 +97,33 @@ def get(self, tag: T.Optional[str]):
         return {"name": tag, "usage_count": tags[1][tag]}
 
     def put(self, tag: str):
+        try:
+            tags = request.data.getlist('tags')  # werkzeug.datastructures.MultiDict
+        except AttributeError:
```

Comment:
When passing `data={'tags': ['tag1', 'tag2']}` using `client.put` in `test_server.py` the received object is of type `werkzeug.datastructures.MultiDict`:
```
request.data.getlist('tags') -> ['tag1', 'tag2']  (list)
request.get('tags') -> 'tag1' (str)
```
However, when passing `data={'tags':['tag1', 'tag2']} using `curl`:
```
request.data.get('tags') -> ['tag1', 'tag2']  (list)
request.data.getlist('tags') -> AttributeError
```

Note that as a side effect ` rd = client.put('/api/tags/old', data={'tags': 'new'})` will return 200 whereas passing the same data with curl will fail with 400...

Reply:
Ah, that's cuz you're _not_ actually submitting JSON with `client.put()` unless you specify it as such, e.g. by [passing the Content-Type header](https://github.com/jarun/buku/blob/v4.8/tests/test_server.py#L214-L215) (what you're sending here instead is `application/x-www-form-urlencoded` with body `tags=tag1&tags=tag2`, which results in a different parser being applied).

I suspect that only JSON is meant to be supported by the API, and thus the tests should be refactored to actually use it (although rather than via custom HTTP headers, I suggest [replacing `data=` with `json=`](https://werkzeug.palletsprojects.com/en/2.2.x/test/#request-body)).